### DISCUSSION TITLE
Crash on load OSX

### DISCUSSION THREAD

**Discussion by r32pat** [ghostty-org/ghostty#3250] [Feature Requests, Ideas] [answered]
Getting an instant crash on load in OSX.
Not related to [https://github.com/ghostty-org/ghostty/issues/3202](url) 

Attempting to launch via Terminal generates the following output.

`/Users\/USER\/Downloads/Ghostty.app/Contents/MacOS/ghostty ; exit;
ghostty(4785,0x170c13000) malloc: *** error for object 0x60000166c000: pointer being freed was not allocated
ghostty(4785,0x170c13000) malloc: *** set a breakpoint in malloc_error_break to debug
Abort trap: 6
logout
`
I have tried installing via homebrew, manually installing the .dmg from the website and tried the nightly build.
All present the same issue.

Can provide full crash log if required.
Anyone else seeing similar issue ?

I'm now currently digging through my installed brew packages to see if something is conflicting somewhere.

**Comment by mitchellh** [comment]
If you can reproduce with a debug build to get a stack trace that'd be excellent. There are no known issues like this currently. And yes, if you can upload crash logs I'll take it.

**Reply by r32pat** [reply]
Using the [ghostty-macos-universal-debug-slow.zip](https://github.com/ghostty-org/ghostty/releases/download/tip/ghostty-macos-universal-debug-slow.zip)

I get a bit more output.
`[sentry] WARN failed to create database directory or there is no write access to this directory
[sentry] WARN `sentry_init` failed
ghostty(9363,0x16e053000) malloc: *** error for object 0x6000039b40c0: pointer being freed was not allocated
ghostty(9363,0x16e053000) malloc: *** set a breakpoint in malloc_error_break to debug
Abort trap: 6`

if you sudo first I get the following.

`sudo ./ghostty 
[sentry] INFO using database path "/Users\/HOME\/.cache/ghostty/sentry"
[sentry] WARN the provided DSN "" is not valid
[sentry] DEBUG starting backend
Usage: ghostty +<action> [flags]

This is the Ghostty helper CLI that accompanies the graphical Ghostty app.
To launch the terminal directly, please launch the graphical app
(i.e. Ghostty.app on macOS). This CLI can be used to perform various
actions such as inspecting the version, listing fonts, etc.

We don't have proper help output yet, sorry! Please refer to the
source code or Discord community for help for now. We'll fix this in time.
`

Running as sudo doesn't crash but also doesn't launch into the GUI as expected.
I have tried adjusting permissions to the .cache directory which moves the issue down the tree by the looks of things.

`./ghostty 
[sentry] INFO using database path "/Users\/HOME\/.cache/ghostty/sentry"
[sentry] WARN failed to lock file "/Users\/HOME\/.cache/ghostty/sentry/b09579af-6211-4be8-79f1-79b1b91f3883.run.lock" (Permission denied)
[sentry] WARN failed to initialize run directory
[sentry] WARN `sentry_init` failed
ghostty(10095,0x16e147000) malloc: *** error for object 0x600003b50000: pointer being freed was not allocated
ghostty(10095,0x16e147000) malloc: *** set a breakpoint in malloc_error_break to debug
Abort trap: 6
`

Is there a particular way you would like me to run it from the command line ?

**Comment by r32pat** [comment, marked as answer]
Answered my own question now and feel a bit stupid about it.
/$HOME/.cache was owned by root:wheel not username:staff
_chown -R username:staff /$HOME/.cache_ fixed the issue.
Just substitute username for your OSX username and should be golden!!

**Reply by r32pat** [reply]
This problem also seemed to be effecting someone on the discord and this solution also worked for them.
Not sure what circumstances cause some OSX users .cache folder to be owned by root instead of their actual user.
But glad we got to the bottom of it.

**Reply by mitchellh** [reply]
Thank you. I updated #3202 to cover both dirs. We should not crash!
